azure/defaults_test.go

```diff
@@ -277,3 +277,74 @@ func TestGetBootstrappingVMExtension(t *testing.T) {
 		})
 	}
 }
+
+func TestNormalizeAzureName(t *testing.T) {
+	testCases := []struct {
+		name     string
+		input    string
+		expected string
+	}{
+		{
```

Comment:
A test case or two to show that this doesn't modify names that are already valid would be good to add.